Mathlib/Topology/Algebra/Module/LinearMapPiProd.lean

```diff
@@ -296,4 +261,74 @@ def prodₗ : ((M →L[R] M₂) × (M →L[R] M₃)) ≃ₗ[S] M →L[R] M₂ ×
 
 end SMul
 
+section coprod
+
+variable {R S M N M₁ M₂ : Type*}
+  [Semiring R] [TopologicalSpace M] [TopologicalSpace N] [TopologicalSpace M₁] [TopologicalSpace M₂]
+
+section AddCommMonoid
+
+variable [AddCommMonoid M] [Module R M] [ContinuousAdd M] [AddCommMonoid N] [Module R N]
+  [ContinuousAdd N] [AddCommMonoid M₁] [Module R M₁] [AddCommMonoid M₂] [Module R M₂]
+
+/-- The continuous linear map given by `(x, y) ↦ f₁ x + f₂ y`. -/
+@[simps! coe apply]
+def coprod (f₁ : M₁ →L[R] M) (f₂ : M₂ →L[R] M) : M₁ × M₂ →L[R] M :=
+  ⟨.coprod f₁ f₂, (f₁.cont.comp continuous_fst).add (f₂.cont.comp continuous_snd)⟩
+
+@[simp] lemma coprod_add (f₁ g₁ : M₁ →L[R] M) (f₂ g₂ : M₂ →L[R] M) :
+    (f₁ + g₁).coprod (f₂ + g₂) = f₁.coprod f₂ + g₁.coprod g₂ := by ext <;> simp
+
+lemma range_coprod (f₁ : M₁ →L[R] M) (f₂ : M₂ →L[R] M) :
+    range (f₁.coprod f₂) = range f₁ ⊔ range f₂ := LinearMap.range_coprod ..
+
+lemma comp_fst_add_comp_snd (f₁ : M₁ →L[R] M) (f₂ : M₂ →L[R] M) :
+    f₁.comp (.fst _ _ _) + f₂.comp (.snd _ _ _) = f₁.coprod f₂ := rfl
+
+lemma comp_coprod (f : M →L[R] N) (g₁ : M₁ →L[R] M) (g₂ : M₂ →L[R] M) :
+    f.comp (g₁.coprod g₂) = (f.comp g₁).coprod (f.comp g₂) :=
+  coe_injective <| LinearMap.comp_coprod ..
+
+@[simp] lemma coprod_comp_inl (f₁ : M₁ →L[R] M) (f₂ : M₂ →L[R] M) :
+    (f₁.coprod f₂).comp (.inl _ _ _) = f₁ := coe_injective <| LinearMap.coprod_inl ..
+
+@[simp] lemma coprod_comp_inr (f₁ : M₁ →L[R] M) (f₂ : M₂ →L[R] M) :
+    (f₁.coprod f₂).comp (.inr _ _ _) = f₂ := coe_injective <| LinearMap.coprod_inr ..
+
+@[simp]
+lemma coprod_inl_inr : ContinuousLinearMap.coprod (.inl R M N) (.inr R M N) = .id R (M × N) :=
+  coe_injective <| LinearMap.coprod_inl_inr
+
+/-- Taking the product of two maps with the same codomain is equivalent to taking the product of
+their domains.
+See note [bundled maps over different rings] for why separate `R` and `S` semirings are used. -/
```

Comment:
``suggestion
/-- Taking the product of two maps with the same codomain is equivalent to taking the product of
their domains.
See note [bundled maps over different rings] for why separate `R` and `S` semirings are used. 

TODO: Upgrade this to a `ContinuousLinearEquiv`. This should be true for any topological 
vector space over a normed field thanks to `ContinuousLinearMap.precomp` and
`ContinuousLinearMap.postcomp`. -/